mlir-tensorrt/compiler/include/mlir-tensorrt/Registration/RegisterMlirTensorRtPasses.h

```diff
@@ -52,6 +53,12 @@ inline void registerAllMlirTensorRtPasses() {
   mlir::registerTransformsPasses();
   mlir::registerConvertPDLToPDLInterp();
 
+  // TODO (pranavm): Check if this needs to be conditional - the TRT passes
+  // above are not.
+#ifdef MLIR_TRT_TARGET_TENSORRT
+  mlirtrt::compiler::registerTensorRTToExecutableTask();
+#endif
```

Comment:
I'm not sure what exactly needs to be guarded. In a lot of places we guard TRT things with `MLIR_TRT_TARGET_TENSORRT` but that doesn't seem to be the case for the TRT passes registered above.

Reply:
The TensorRT dialect and MLIR passes are always built. It doesn't depend on actually having TensorRT binaries or headers to build against (or at least, that was the idea, we haven't been enforcing it). The `MLIR_TRT_TARGET_TENSORRT` is basically a guard for anything  that relies on having actual TRT headers or libraries to link against, for example the translation from MLIR to TensorRT, hence the `TARGET_TENSORRT` in the name.